File: agbac-min/agents/typescript/adapters/auth0Adapter.ts
/**
 * auth0Adapter.ts
 * Implements AdapterBase for Auth0.
 */

import { AdapterBase } from './adapterBase';
import axios from 'axios';

export class Auth0Adapter extends AdapterBase {
    private tokenUrl: string;
    private clientId: string;
    private clientSecret: string;

    constructor(tokenUrl: string, clientId: string, clientSecret: string) {
        super();
        this.tokenUrl = tokenUrl;
        this.clientId = clientId;
        this.clientSecret = clientSecret;
    }

    async requestToken(act: { sub: string }, sub: string): Promise<string> {
        try {
            this.logInfo('Requesting token from Auth0 with sub and act claims');

            const response = await axios.post(this.tokenUrl, {
                grant_type: 'client_credentials',
                client_id: this.clientId,
                client_secret: this.clientSecret,
                audience: 'https://api.example.com',
                sub: sub,
                act: act
            }, {
                headers: { 'Content-Type': 'application/json' }
            });

            return response.data.access_token;
        } catch (error) {
            this.handleError(error, 'Auth0Adapter.requestToken');
        }
    }
}

File: agbac-min/agents/python/adapters/auth0_adapter.py
import logging
import requests
from .base_adapter import BaseAdapter

logger = logging.getLogger(__name__)
logger.setLevel(logging.INFO)

class Auth0Adapter(BaseAdapter):
    """
    Adapter for Auth0 OAuth2 token requests.
    Supports dual-subject tokens with sub (agent) and act (human).
    """

    def __init__(self, domain: str, client_id: str, client_secret: str):
        self.token_url = f"https://{domain}/oauth/token"
        self.client_id = client_id
        self.client_secret = client_secret

    def request_token(self, sub: str, act: dict, scope: list) -> dict:
        payload = {
            "grant_type": "client_credentials",
            "client_id": self.client_id,
            "client_secret": self.client_secret,
            "audience": "https://api.example.com/",
            "scope": " ".join(scope),
            "sub": sub,
            "act": act
        }

        try:
            response = requests.post(self.token_url, json=payload)
            response.raise_for_status()
            token_data = response.json()
            logger.info(f"Auth0 token issued for agent {sub} with human act {act}")
            return token_data
        except requests.HTTPError as e:
            logger.error(f"Auth0 token request failed: {e}")
            raise
        except Exception as e:
            logger.error(f"Unexpected error in Auth0Adapter: {e}")
            raise
